src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -250,6 +250,115 @@ public static void IgnoredDataShouldNotBeExtensionData()
+        }
+
+        private class ClassWithExtensionDataWithAttributedConverter
+        {
+            [JsonExtensionData]
+            [JsonConverter(typeof(DictionaryOverflowConverter))]
+            public Dictionary<string, object> Overflow { get; set; }
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_SupportsWritingToAttributedCustomSerializer()
+        {
+            var root = new ClassWithExtensionDataWithAttributedConverter
+            {
+                Overflow = new Dictionary<string, object>
+                {
+                    ["TestKey"] = "TestValue",
+                },
+            };
+
+            string json = JsonSerializer.Serialize(root);
+            Assert.Equal(@"{""MyCustomOverflowWrite"":""OverflowValueWrite""}", json);
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_IgnoresCustomSerializerWithOptions()
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new DictionaryOverflowConverter());
+
+            ClassWithExtensionData obj
+                = JsonSerializer.Deserialize<ClassWithExtensionData>(@"{""TestKey"":""TestValue""}", options);
+
+            Assert.Equal("TestValue", ((JsonElement)obj.Overflow["TestKey"]).GetString());
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_IgnoresExplicitCustomSerializer()
+        {
+            ClassWithExtensionDataWithAttributedConverter obj
+                = JsonSerializer.Deserialize<ClassWithExtensionDataWithAttributedConverter>(@"{""TestKey"":""TestValue""}");
+
+            Assert.Equal("TestValue", ((JsonElement)obj.Overflow["TestKey"]).GetString());
+        }
+
+        private class ClassWithJsonElementExtensionData
+        {
+            [JsonExtensionData]
+            [JsonConverter(typeof(JsonElementOverflowConverter))]
+            public Dictionary<string, JsonElement> Overflow { get; set; }
+        }
+
+        public class JsonElementOverflowConverter : JsonConverter<Dictionary<string, JsonElement>>
+        {
+            public override Dictionary<string, JsonElement> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, Dictionary<string, JsonElement> value, JsonSerializerOptions options)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [Fact]
+        public static void ExtensionPropertyWithJsonElement_IgnoresExplicitCustomSerializer()
+        {
+            ClassWithJsonElementExtensionData obj
+                = JsonSerializer.Deserialize<ClassWithJsonElementExtensionData>(@"{""TestKey"":""TestValue""}");
+
+            Assert.Equal("TestValue", obj.Overflow["TestKey"].GetString());
```

Comment:
We should test serialization here (`Assert.Throws` is fine).

Reply:
The new test covers both serialization and deserialization